geps/gep-1619.md

```diff
@@ -163,43 +163,258 @@ Java application servers such as Tomcat and Jetty, were the first to standardize
 
 ### Session Persistence API
 
-The majority of implementations and dataplanes support configuring cookies for session persistence while support for header-based session persistence is limited.
+Although some implementations and dataplanes support session persistence through headers, cookies are more universally
+supported and portable; therefore, this GEP will focus on implementing cookie-based session persistence.
 
-To organize potential Gateway API configuration of cookie functionality, the next section will break down configuration into specific categories:
-- Session Cookie Configuration
-- Session Cookie Strategy
+To determine how the session persistence API should be designed, we must consider the following:
+1. Who is responsible for configuring it?
+2. What existing API object it is augmenting?
+3. What are the practical implications of attaching session persistence to an existing API object?
+4. How are existing implementation APIs designed?
 
-**Session Cookie Configuration**
+**Who is responsible for configuring it?**
 
-Session cookie configuration are the parameters and attributes the dataplane uses to create a cookie for establishing a session.
+Referring to the [Gateway API Security Model](https://gateway-api.sigs.k8s.io/concepts/security-model/#roles-and-personas),
+the target kubernetes role/persona for session persistence, are application developers. In most cases, session
+persistence is not considered an optional feature, but rather a necessary and deliberate component incorporated into the
+application's architecture by the developers themselves. Consequently, application developers should be responsible for
+adjusting the configuration to ensure the functionality of their applications.
 
-A set-cookie request can be broken down into [3 main components](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie):
-- Cookie Name
-- Cookie Value
-- Cookie Attributes
+**What existing API object it is augmenting?**
 
-Only the Cookie Name and Cookie Value are required. All Cookie Attributes are optional. Attributes are simply key=value pairs with the value as optional for some attributes.
+Session persistence augments a Kubernetes service object after the routing to the service has taken place. In
+Kubernetes, a service is an abstract representation of a set of pods that work together to provide a specific
+functionality. Services perform load balancing, routing traffic from a service endpoint to a pod. Session persistence
+augments the load balancing functionality of the service by consistently directing traffic to the same pod.
```

Comment:
"to the same pod" and to the same service containing that pod. Just to reiterate my point in https://github.com/kubernetes-sigs/gateway-api/pull/2159/files#r1252413412

Reply:
Right, I updated this section to reflect that.

Reply:
I'm not entirely sure this belong to the spec or is a requirement.  For implementations that encode the IP - it's a must because of IP reuse. For implementations that encode a durable pod ID - this may be sufficient. 

There are users who dynamically change label selector on a service, so a Pod may start selected by canary-service and then be selected by prod-service (without restarting the pod). This model is not compatible with current istio implementation of persistent sessions for example, but it's valid and used by some CI/CD tools.

I would leave this to implementations.

Reply:
@costinm what is "this" in the context of you comment? Are you referring to Sanjay's statement on clarifying `"to the same pod" and to the same service containing that pod.`?

Are you saying we shouldn't specify how traffic splitting affects session persistence and leave that up to implementations?